conf.h file not found

### Acton Version

0.26.0.20260211.10.52

### Steps to Reproduce and Observed Behavior

I deleted a `.act` file I longer wanted to use / be included elsewhere in my project, but an import statement refering to that file still existed (even if nothing from that import was being used).

Rather than getting a nice Acton compiler error telling me about the missing module I got this:

```
one-fabric-orch on  main [✘!] via ↯ took 8s
❯ make build
acton build
Building project in /Users/klambrec/Developer/autlan/one-fabric-orch
Resolving dependencies (fetching if missing)...
Using cached pkg dependency actmf (N-V-__8AAJtaBABOb7WyjNcnWG3420pPMwvehI80qJohWsmC)
Using cached pkg dependency netconf (N-V-__8AAOL3AQD7Oy0qj2nAbYZ0u3dGWSGfvqABuALB3K0y)
Using cached pkg dependency orchestron (N-V-__8AAFBUEQBjXaW-92KIz_iNxo4_QeS9zuIj8QyUujZO)
Using cached pkg dependency rtrsecrets (N-V-__8AADYUAAD8XrwSuK5_7k2ua6Z05J5u-UIcoEhLNsmh)
Using cached pkg dependency yang (N-V-__8AAIOYIABUmXlnf1AZF-mtRvZcsWOi3MzODQhXHGxf)
 ⠇ Running final compilation:                                                0.566 s
ERROR: internal compiler error: compilation of generated Zig code failed, returned error code1
NOTE: this is likely a bug in acton, please report this at:
NOTE: https://github.com/actonlang/acton/issues/new?template=ice.yaml
NOTE: acton 0.26.0.20260211.10.52
zig stdout:

zig stderr:
Acton Project Builder - building /Users/klambrec/Developer/autlan/one-fabric-orch
Acton Base Builder
Building in /Users/klambrec/.local/acton/base
-- filename : json.c
   full_path: /Users/klambrec/.local/acton/base/out/types/json.c
   dir      : /Users/klambrec/.local/acton/base/out/types
   file_path: /json.c
-- filename : xml.c
   full_path: /Users/klambrec/.local/acton/base/out/types/xml.c
   dir      : /Users/klambrec/.local/acton/base/out/types
   file_path: /xml.c
-- filename : diff.c
   full_path: /Users/klambrec/.local/acton/base/out/types/diff.c
   dir      : /Users/klambrec/.local/acton/base/out/types
   file_path: /diff.c
-- filename : md5.c
   full_path: /Users/klambrec/.local/acton/base/out/types/crypto/hash/md5.c
   dir      : /Users/klambrec/.local/acton/base/out/types/crypto/hash
   file_path: /crypto/hash/md5.c
-- filename : term.c
   full_path: /Users/klambrec/.local/acton/base/out/types/term.c
   dir      : /Users/klambrec/.local/acton/base/out/types
   file_path: /term.c
-- filename : process.c
   full_path: /Users/klambrec/.local/acton/base/out/types/process.c
   dir      : /Users/klambrec/.local/acton/base/out/types
   file_path: /process.c
-- filename : math.c
   full_path: /Users/klambrec/.local/acton/base/out/types/math.c
   dir      : /Users/klambrec/.local/acton/base/out/types
   file_path: /math.c
-- filename : wyhash.c
   full_path: /Users/klambrec/.local/acton/base/out/types/hash/wyhash.c
   dir      : /Users/klambrec/.local/acton/base/out/types/hash
   file_path: /hash/wyhash.c
-- filename : random.c
   full_path: /Users/klambrec/.local/acton/base/out/types/random.c
   dir      : /Users/klambrec/.local/acton/base/out/types
   file_path: /random.c
-- filename : fs.c
   full_path: /Users/klambrec/.local/acton/base/out/types/fs.c
   dir      : /Users/klambrec/.local/acton/base/out/types
   file_path: /fs.c
-- filename : qcheck.c
   full_path: /Users/klambrec/.local/acton/base/out/types/qcheck.c
   dir      : /Users/klambrec/.local/acton/base/out/types
   file_path: /qcheck.c
-- filename : __builtin__.c
   full_path: /Users/klambrec/.local/acton/base/out/types/__builtin__.c
   dir      : /Users/klambrec/.local/acton/base/out/types
   file_path: /__builtin__.c
-- filename : time.c
   full_path: /Users/klambrec/.local/acton/base/out/types/time.c
   dir      : /Users/klambrec/.local/acton/base/out/types
   file_path: /time.c
-- filename : testing.c
   full_path: /Users/klambrec/.local/acton/base/out/types/testing.c
   dir      : /Users/klambrec/.local/acton/base/out/types
   file_path: /testing.c
-- filename : argparse.c
   full_path: /Users/klambrec/.local/acton/base/out/types/argparse.c
   dir      : /Users/klambrec/.local/acton/base/out/types
   file_path: /argparse.c
-- filename : base64.c
   full_path: /Users/klambrec/.local/acton/base/out/types/base64.c
   dir      : /Users/klambrec/.local/acton/base/out/types
   file_path: /base64.c
-- filename : net.c
   full_path: /Users/klambrec/.local/acton/base/out/types/net.c
   dir      : /Users/klambrec/.local/acton/base/out/types
   file_path: /net.c
-- filename : http.c
   full_path: /Users/klambrec/.local/acton/base/out/types/http.c
   dir      : /Users/klambrec/.local/acton/base/out/types
   file_path: /http.c
-- filename : file.c
   full_path: /Users/klambrec/.local/acton/base/out/types/file.c
   dir      : /Users/klambrec/.local/acton/base/out/types
   file_path: /file.c
-- filename : re.c
   full_path: /Users/klambrec/.local/acton/base/out/types/re.c
   dir      : /Users/klambrec/.local/acton/base/out/types
   file_path: /re.c
-- filename : buildy.c
   full_path: /Users/klambrec/.local/acton/base/out/types/buildy.c
   dir      : /Users/klambrec/.local/acton/base/out/types
   file_path: /buildy.c
-- filename : uri.c
   full_path: /Users/klambrec/.local/acton/base/out/types/uri.c
   dir      : /Users/klambrec/.local/acton/base/out/types
   file_path: /uri.c
-- filename : rts.c
   full_path: /Users/klambrec/.local/acton/base/out/types/acton/rts.c
   dir      : /Users/klambrec/.local/acton/base/out/types/acton
   file_path: /acton/rts.c
-- filename : snappy.c
   full_path: /Users/klambrec/.local/acton/base/out/types/snappy.c
   dir      : /Users/klambrec/.local/acton/base/out/types
   file_path: /snappy.c
-- filename : logging.c
   full_path: /Users/klambrec/.local/acton/base/out/types/logging.c
   dir      : /Users/klambrec/.local/acton/base/out/types
   file_path: /logging.c
Debug build
Threads enabled
Acton Project Builder - building /Users/klambrec/.cache/acton/deps/actmf-N-V-__8AAJtaBABOb7WyjNcnWG3420pPMwvehI80qJohWsmC
Acton Project Builder - building /Users/klambrec/.cache/acton/deps/yang-N-V-__8AAIOYIABUmXlnf1AZF-mtRvZcsWOi3MzODQhXHGxf
Debug build
Threads enabled
Threads enabled
Debug build
Threads enabled
Acton Project Builder - building /Users/klambrec/.cache/acton/deps/netconf-N-V-__8AAOL3AQD7Oy0qj2nAbYZ0u3dGWSGfvqABuALB3K0y
Debug build
Threads enabled
Acton Project Builder - building /Users/klambrec/.cache/acton/deps/orchestron-N-V-__8AAFBUEQBjXaW-92KIz_iNxo4_QeS9zuIj8QyUujZO
Debug build
Threads enabled
Acton Project Builder - building /Users/klambrec/.cache/acton/deps/rtrsecrets-N-V-__8AADYUAAD8XrwSuK5_7k2ua6Z05J5u-UIcoEhLNsmh
Debug build
Threads enabled
Debug build
Threads enabled
Building executable from: /Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric.test_root.c -> .test_one_fabric
Building executable from: /Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric.root.c -> one_fabric
install
+- install .test_one_fabric
   +- compile exe .test_one_fabric Debug aarch64-macos-none
      +- compile lib ActonProject Debug aarch64-macos-none 4 errors
/Users/klambrec/Developer/autlan/one-fabric-orch/./out/types/one_fabric/cfs.h:8:10: error: 'out/types/one_fabric/conf.h' file not found
#include "out/types/one_fabric/conf.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric/cfs.c:3:10: note: in file included from /Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric/cfs.c:3:
#include "out/types/one_fabric/cfs.h"
         ^
/Users/klambrec/Developer/autlan/one-fabric-orch/./out/types/one_fabric/cfs.h:8:10: error: 'out/types/one_fabric/conf.h' file not found
#include "out/types/one_fabric/conf.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric/layers.c:3:10: note: in file included from /Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric/layers.c:3:
#include "out/types/one_fabric/layers.h"
         ^
/Users/klambrec/Developer/autlan/one-fabric-orch/./out/types/one_fabric/layers.h:12:10: note: in file included from /Users/klambrec/Developer/autlan/one-fabric-orch/./out/types/one_fabric/layers.h:12:
#include "out/types/one_fabric/layers/t_0.h"
         ^
/Users/klambrec/Developer/autlan/one-fabric-orch/./out/types/one_fabric/layers/t_0.h:9:10: note: in file included from /Users/klambrec/Developer/autlan/one-fabric-orch/./out/types/one_fabric/layers/t_0.h:9:
#include "out/types/one_fabric/cfs.h"
         ^
/Users/klambrec/Developer/autlan/one-fabric-orch/./out/types/one_fabric/cfs.h:8:10: error: 'out/types/one_fabric/conf.h' file not found
#include "out/types/one_fabric/conf.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric/layers/t_0.c:3:10: note: in file included from /Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric/layers/t_0.c:3:
#include "out/types/one_fabric/layers/t_0.h"
         ^
/Users/klambrec/Developer/autlan/one-fabric-orch/./out/types/one_fabric/layers/t_0.h:9:10: note: in file included from /Users/klambrec/Developer/autlan/one-fabric-orch/./out/types/one_fabric/layers/t_0.h:9:
#include "out/types/one_fabric/cfs.h"
         ^
/Users/klambrec/Developer/autlan/one-fabric-orch/./out/types/one_fabric/cfs.h:8:10: error: 'out/types/one_fabric/conf.h' file not found
#include "out/types/one_fabric/conf.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric.c:3:10: note: in file included from /Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric.c:3:
#include "out/types/one_fabric.h"
         ^
/Users/klambrec/Developer/autlan/one-fabric-orch/./out/types/one_fabric.h:23:10: note: in file included from /Users/klambrec/Developer/autlan/one-fabric-orch/./out/types/one_fabric.h:23:
#include "out/types/one_fabric/layers.h"
         ^
/Users/klambrec/Developer/autlan/one-fabric-orch/./out/types/one_fabric/layers.h:12:10: note: in file included from /Users/klambrec/Developer/autlan/one-fabric-orch/./out/types/one_fabric/layers.h:12:
#include "out/types/one_fabric/layers/t_0.h"
         ^
/Users/klambrec/Developer/autlan/one-fabric-orch/./out/types/one_fabric/layers/t_0.h:9:10: note: in file included from /Users/klambrec/Developer/autlan/one-fabric-orch/./out/types/one_fabric/layers/t_0.h:9:
#include "out/types/one_fabric/cfs.h"
         ^
error: the following command failed with 4 compilation errors:
/Users/klambrec/.local/acton/zig/zig build-lib -cflags -ffile-prefix-map=/Users/klambrec/Developer/autlan/= -DDEV -DACTON_THREADS -fno-sanitize=signed-integer-overflow -- /Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric.c -cflags -ffile-prefix-map=/Users/klambrec/Developer/autlan/= -DDEV -DACTON_THREADS -fno-sanitize=signed-integer-overflow -- /Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric/layers/y_1.c -cflags -ffile-prefix-map=/Users/klambrec/Developer/autlan/= -DDEV -DACTON_THREADS -fno-sanitize=signed-integer-overflow -- /Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric/layers/t_0.c -cflags -ffile-prefix-map=/Users/klambrec/Developer/autlan/= -DDEV -DACTON_THREADS -fno-sanitize=signed-integer-overflow -- /Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric/layers/base_0.c -cflags -ffile-prefix-map=/Users/klambrec/Developer/autlan/= -DDEV -DACTON_THREADS -fno-sanitize=signed-integer-overflow -- /Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric/layers/y_2.c -cflags -ffile-prefix-map=/Users/klambrec/Developer/autlan/= -DDEV -DACTON_THREADS -fno-sanitize=signed-integer-overflow -- /Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric/layers/base_1.c -cflags -ffile-prefix-map=/Users/klambrec/Developer/autlan/= -DDEV -DACTON_THREADS -fno-sanitize=signed-integer-overflow -- /Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric/layers/y_0_loose.c -cflags -ffile-prefix-map=/Users/klambrec/Developer/autlan/= -DDEV -DACTON_THREADS -fno-sanitize=signed-integer-overflow -- /Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric/layers/y_1_loose.c -cflags -ffile-prefix-map=/Users/klambrec/Developer/autlan/= -DDEV -DACTON_THREADS -fno-sanitize=signed-integer-overflow -- /Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric/layers/t_2.c -cflags -ffile-prefix-map=/Users/klambrec/Developer/autlan/= -DDEV -DACTON_THREADS -fno-sanitize=signed-integer-overflow -- /Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric/layers/base_2.c -cflags -ffile-prefix-map=/Users/klambrec/Developer/autlan/= -DDEV -DACTON_THREADS -fno-sanitize=signed-integer-overflow -- /Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric/layers/y_0.c -cflags -ffile-prefix-map=/Users/klambrec/Developer/autlan/= -DDEV -DACTON_THREADS -fno-sanitize=signed-integer-overflow -- /Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric/layers/t_1.c -cflags -ffile-prefix-map=/Users/klambrec/Developer/autlan/= -DDEV -DACTON_THREADS -fno-sanitize=signed-integer-overflow -- /Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric/layers/y_2_loose.c -cflags -ffile-prefix-map=/Users/klambrec/Developer/autlan/= -DDEV -DACTON_THREADS -fno-sanitize=signed-integer-overflow -- /Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric/layers.c -cflags -ffile-prefix-map=/Users/klambrec/Developer/autlan/= -DDEV -DACTON_THREADS -fno-sanitize=signed-integer-overflow -- /Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric/sysspec.c -cflags -ffile-prefix-map=/Users/klambrec/Developer/autlan/= -DDEV -DACTON_THREADS -fno-sanitize=signed-integer-overflow -- /Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric/rfs.c -cflags -ffile-prefix-map=/Users/klambrec/Developer/autlan/= -DDEV -DACTON_THREADS -fno-sanitize=signed-integer-overflow -- /Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric/tmf.c -cflags -ffile-prefix-map=/Users/klambrec/Developer/autlan/= -DDEV -DACTON_THREADS -fno-sanitize=signed-integer-overflow -- /Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric/cfs.c -cflags -ffile-prefix-map=/Users/klambrec/Developer/autlan/= -DDEV -DACTON_THREADS -fno-sanitize=signed-integer-overflow -- /Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric/devices/JuniperCRPD_23_4R1_9.c -cflags -ffile-prefix-map=/Users/klambrec/Developer/autlan/= -DDEV -DACTON_THREADS -fno-sanitize=signed-integer-overflow -- /Users/klambrec/Developer/autlan/one-fabric-orch/out/types/one_fabric/devices/CiscoIosXr_24_1_ncs55a1.c -ODebug -target aarch64-macos-none -mcpu apple_m1 -I /Users/klambrec/Developer/autlan/one-fabric-orch/. -I /Users/klambrec/.cache/acton/zig-local-cache/o/6446de79bb460ada0cb5609c27e81247 -I /Users/klambrec/.cache/acton/zig-local-cache/o/8821d4ef51c7757e9eb636906d823170 -I /Users/klambrec/.cache/acton/zig-local-cache/o/d9020562b1bb8c910441c78f37bf9ef4 -I /Users/klambrec/.cache/acton/zig-local-cache/o/16316aabea5bd78754f14ecfee698c89 -I /Users/klambrec/.cache/acton/zig-local-cache/o/d47702a77730caf0615548fd8997a52f -I /Users/klambrec/.cache/acton/zig-local-cache/o/0701ce2ce10bc6639a3ae524fec711c0 -Mroot -lc++ -lc --cache-dir /Users/klambrec/.cache/acton/zig-local-cache --global-cache-dir /Users/klambrec/.cache/acton/zig-global-cache --name ActonProject -static --zig-lib-dir /Users/klambrec/.local/acton/zig/lib/ --listen=-

Build Summary: 46/53 steps succeeded; 1 failed
install transitive failure
+- install ActonProject transitive failure
|  +- compile lib ActonProject Debug aarch64-macos-none 4 errors
+- install .test_one_fabric transitive failure
|  +- compile exe .test_one_fabric Debug aarch64-macos-none transitive failure
|     +- compile lib ActonProject Debug aarch64-macos-none (+18 more reused dependencies)
|     +- compile lib ActonProject Debug aarch64-macos-none (+18 more reused dependencies)
+- install one_fabric transitive failure
   +- compile exe one_fabric Debug aarch64-macos-none transitive failure
      +- compile lib ActonProject Debug aarch64-macos-none (+18 more reused dependencies)
      +- compile lib ActonProject Debug aarch64-macos-none (+18 more reused dependencies)

error: the following build command failed with exit code 1:
/Users/klambrec/.cache/acton/zig-local-cache/o/a164f54f04c68839a2269e0e5567441e/build /Users/klambrec/.local/acton/zig/zig /Users/klambrec/.local/acton/zig/lib /Users/klambrec/Developer/autlan/one-fabric-orch /Users/klambrec/.cache/acton/zig-local-cache /Users/klambrec/.cache/acton/zig-global-cache --seed 0xf575c4b9 -Zc777bfbbc00f98b9 --prefix /Users/klambrec/Developer/autlan/one-fabric-orch/out --prefix-exe-dir bin -Dtarget=aarch64-macos-none -Dcpu=apple_m1 -Doptimize=Debug
make: *** [build] Error 1
```

### Expected Behavior

I expect a nice actonc error, this should be easily reproducible, e.g. in `sorespo` by removing `conf.act`, and replacing e.g.

`        fm = o.netinfra.ibgp_fullmesh.create(i.asn, IGBP_AUTHENTICATION_KEY)`

with:
`        fm = o.netinfra.ibgp_fullmesh.create(i.asn, "test")`

But leaving the relevant import statement in place.

--- a/compiler/acton/test_incremental.hs
+++ b/compiler/acton/test_incremental.hs
@@ -10,6 +10,7 @@ import qualified Data.Text.IO as T
 import qualified Data.Text.Encoding as TE
 import           Data.Char (isDigit, isHexDigit, isSpace)
 import           Data.List (find, partition, sort)
+import qualified Data.Map.Strict as M
 import           System.Directory
 import           System.Exit
 import           System.FilePath
@@ -1555,6 +1556,62 @@ p34_removed_import_module = testCase "34-removed imported module fails before Zi
     (T.isInfixOf "Type interface file not found or unreadable for a" out)
   assertBool "did not expect Zig build to run" (not (T.isInfixOf "zigCmd:" out))
 
+p35_changed_path_keeps_unaffected_provider :: TestTree
+p35_changed_path_keeps_unaffected_provider =
+  testCase "35-changed-path incremental keeps unchanged providers" $ do
+    let proj = casesProjDir
+        src = casesSrcDir
+        actA = src </> "a.act"
+        actB = src </> "b.act"
+        actC = src </> "c.act"
+        actMain = src </> "main.act"
+        gopts = C.GlobalOptions
+          { C.color = C.Never
+          , C.quiet = True
+          , C.timing = False
+          , C.tty = False
+          , C.verbose = False
+          , C.verboseZig = False
+          , C.jobs = 1
+          }
+        opts = Compile.defaultCompileOptions { C.only_build = True }
+    ensureCasesProject
+    writeFileUtf8 actA "aaa = 1\n"
+    writeFileUtf8 actC "ccc = 10\n"
+    writeFileUtf8 actB $ T.unlines
+      [ "import a"
+      , "import c"
+      , ""
+      , "def bar() -> int:"
+      , "    return a.aaa + c.ccc"
+      ]
+    writeFileUtf8 actMain $ T.unlines
+      [ "import b"
+      , ""
+      , "actor main(env: Env):"
+      , "    print(b.bar())"
+      , "    env.exit(0)"
+      ]
+    _ <- buildOutIn proj
+    writeFileUtf8 actA "aaa = 2\n"
+    actAAbs <- canonicalizePath actA
+    actMainAbs <- canonicalizePath actMain
+    sched <- Compile.newCompileScheduler gopts 1
+    plan <- Compile.prepareCompilePlan
+      Source.diskSourceProvider
+      gopts
+      sched
+      opts
+      [actMainAbs]
+      False
+      (Just [actAAbs])
+    bTask <- case find (\t -> Compile.name (Compile.gtTask t) == A.modName ["b"]) (Compile.cpNeededTasks plan) of
+      Just t -> pure t
+      Nothing -> assertFailure "expected b in changed-path compile plan" >> fail "missing b task"
+    let bProviders = Compile.gtImportProviders bTask
+    assertBool "expected provider for changed import a" (M.member (A.modName ["a"]) bProviders)
+    assertBool "expected provider for unchanged import c" (M.member (A.modName ["c"]) bProviders)
+
 -- Main -----------------------------------------------------------------------
 
 -- | Tasty entry point for incremental tests.
@@ -1609,5 +1666,6 @@ main = defaultMain $ localOption (NumThreads 1) $ testGroup "incremental"
       , p32_project_alt_output
       , p33_comprehensive_hashes
       , p34_removed_import_module
+      , p35_changed_path_keeps_unaffected_provider
       ]
   ]

--- a/compiler/lib/src/Acton/Compile.hs
+++ b/compiler/lib/src/Acton/Compile.hs
@@ -1061,9 +1061,9 @@ selectAffectedTasks globalTasks changedFiles = do
                                 M.empty
                                 (M.toList depMap)
                 affected = reverseClosure revMap (Data.Set.fromList changedKeys)
-                keepProviders t =
-                  t { gtImportProviders = M.filter (`Data.Set.member` affected) (gtImportProviders t) }
-            return [ keepProviders t | t <- globalTasks, Data.Set.member (gtKey t) affected ]
+            -- Keep original provider mappings so unchanged imports can still
+            -- resolve via cached interfaces during incremental checks.
+            return [ t | t <- globalTasks, Data.Set.member (gtKey t) affected ]
   where
     reverseClosure revMap start = go start (Data.Set.toList start)
       where
@@ -1891,14 +1891,20 @@ compileTasks sp gopts opts rootPaths rootProj tasks callbacks = do
               Just depKey ->
                 case M.lookup depKey pubMap of
                   Just h  -> return (Just h)
-                  Nothing -> error ("Internal error: missing pub hash for dep " ++ modNameToString m)
+                  Nothing ->
+                    if M.member depKey taskMap
+                      then error ("Internal error: missing pub hash for dep " ++ modNameToString m)
+                      else getPubHashCached paths m
               Nothing -> getPubHashCached paths m
           resolveNameHashMap' m =
             case M.lookup m providers of
               Just depKey ->
                 case M.lookup depKey nameMap of
                   Just hm -> return (Just hm)
-                  Nothing -> error ("Internal error: missing name hashes for dep " ++ modNameToString m)
+                  Nothing ->
+                    if M.member depKey taskMap
+                      then error ("Internal error: missing name hashes for dep " ++ modNameToString m)
+                      else getNameHashMapCached paths m
               Nothing -> getNameHashMapCached paths m
 
           missingNameHashDiagnostics qn =

